mathesar/api/db/viewsets/records.py

```diff
@@ -103,11 +103,16 @@ def list(self, request, table_pk=None):
                 status_code=status.HTTP_400_BAD_REQUEST
             )
         except DataError as e:
-            if isinstance(e.orig, InvalidDatetimeFormat):
+            if isinstance(e.orig, (InvalidDatetimeFormat, DatetimeFieldOverflow)):
```

Comment:
Please split these to raise `InvalidDateFormatAPIException` and `InvalidDateAPIException` respectively.